src/run/run_environment/github_actions/provider.rs

```diff
@@ -236,6 +245,46 @@ impl RunEnvironmentProvider for GitHubActionsProvider {
             .to_string();
         Ok(commit_hash)
     }
+
+    /// Get the OIDC token for GitHub Actions.
+    ///
+    /// This requires that the workflow has the `id-token` permission enabled.
+    ///
+    /// Docs:
+    /// - https://docs.github.com/en/actions/how-tos/secure-your-work/security-harden-deployments/oidc-with-reusable-workflows
+    /// - https://docs.github.com/en/actions/concepts/security/openid-connect
+    /// - https://docs.github.com/en/actions/reference/security/oidc#methods-for-requesting-the-oidc-token
+    async fn get_oidc_token(&self) -> Option<String> {
+        // The `ACTIONS_ID_TOKEN_REQUEST_TOKEN` environment variable is set when the `id-token` permission is granted.
+        // This is necessary to authenticate with OIDC, but not strictly set just for OIDC.
+        let request_token = get_env_variable("ACTIONS_ID_TOKEN_REQUEST_TOKEN").ok();
+        let request_url = get_env_variable("ACTIONS_ID_TOKEN_REQUEST_URL").ok();
+
+        if request_token.is_none() || request_url.is_none() {
+            warn!(
+                "The Workflow is missing the 'id-token: write' permission required to request an OIDC token. \
+                Please refer to https://docs.github.com/en/actions/how-to-guides/security-hardening-your-workflows/about-security-hardening-with-openid-connect for more information."
+            );
+
+            return None;
+        }
+
+        let request_url = request_url.unwrap();
+        let request_url = format!("{request_url}&audience=codspeed");
+        let request_token = request_token.unwrap();
+
+        Client::new()
```

Comment:
Let's add a similar retrying logic as they do in the OIDC client in https://github.com/actions/toolkit/blob/dcae869a030cef8949f37078ccc240752fb69338/packages/core/src/oidc-utils.ts#L14